src/server/snapshot.h

```diff
@@ -19,6 +22,22 @@ struct Entry;
 
 class RdbSerializer;
 
+class ZstdCompress {
```

Comment:
nit: i would move this class definition into cc and just forward declare it here. 
why - to avoid bringing "zstd.h" includes to all the cc files that include this header since it's really an implementation detail.  Also see my comment below about unique_ptr